src/app/fact-page/fact-page.component.ts

```diff
@@ -0,0 +1,51 @@
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+import { NumberApiServiceService } from '../numbers-api-service/number-api-service.service';
+import { ApiResponse } from '../app.model';
+import { HttpErrorResponse } from '@angular/common/http';
+
+@Component({
+  selector: 'app-fact-page',
+  templateUrl: './fact-page.component.html',
+  styleUrls: ['./fact-page.component.css']
+})
+export class FactPageComponent implements OnInit, OnDestroy {
+  public processing = true;
+  public numberFact: ApiResponse;
+  public networkError: HttpErrorResponse;
+  private readonly onDestroy$ = new Subject<void>();
+
+  constructor(
+    private route: ActivatedRoute,
+    private router: Router,
+    private service: NumberApiServiceService,
+  ) { }
+
+  public ngOnInit() {
+    this.route.params
+      .pipe(takeUntil(this.onDestroy$))
+      .subscribe(params => {
+        if (isNaN(params.id)) {
+          this.router.navigateByUrl('/');
+        } else {
+          this.service.getFact(params.id)
+            .pipe(takeUntil(this.onDestroy$))
```

Comment:
Not needed here (see comments for `InputContainer`)

Reply:
Will change